### PR TITLE
Add FXIOS-10715 [sponsored tiles] Get unified ads data with a POST request

### PR DESCRIPTION
**Pull request by lmarceau**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-10715)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/23418)

## :bulb: Description
- Made changes to `UnifiedAdsProvider` to build a request either on the prod or staging endpoint. I kept the same caching mechanism as with `ContileProvider`, so tiles will be cached for an hour. 
- Removed `UnifiedAdsNetwork` since turns out I can use `ContileNetworking` to make the url session request.
- Note; this code is meant to be short lived. This API will be moved to A-S next year, so aiming to do a quick solution using a similar approach to what currently exist in the application.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [X] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift [hunk 1]
```diff
@@ -6,21 +6,38 @@ import Common
...
+    func fetchTiles(timestamp: Timestamp = Date.now(), completion: @escaping (UnifiedTileResult) -> Void) {
+        fetchTiles(timestamp: timestamp, completion: completion)
+    }
 }
 
-class UnifiedAdsProvider: UnifiedAdsProviderInterface {
-    private static let resourceEndpoint = "https://ads.mozilla.org/v1/ads"
+class UnifiedAdsProvider: URLCaching, UnifiedAdsProviderInterface, FeatureFlaggable {
+    private static let prodResourceEndpoint = "https://ads.mozilla.org/v1/ads"
+    static let stagingResourceEndpoint = "https://ads.allizom.org/v1/ads"
```
Comment (adudenamedruby) [inline firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift:29]: are we sure we want the ads urls public? Has this been discussed?

**Review by adudenamedruby — APPROVED** [review]
@OrlaM is this the UnifiedAds work we were saying would be a good candidate to live in AS?

File: firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift (hunk 1 above)
Comment (lmarceau) [reply firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift:29]: Mhh, so this one is public as the one is public in `ContileProvider`. It's public to be accessed by the unit tests. It's not a good reasons to make it public, so happy to change this. I guess I could duplicate it for unit tests if it's not public?

Comment (adudenamedruby) [reply firefox-ios/Client/Frontend/Home/TopSites/DataManagement/UnifiedAds/UnifiedAdsProvider.swift:29]: Just asking that we're ok with that. For wallpapers, for example, it was explicitly not made public, so I'm just checking.

**Comment by adudenamedruby** [comment]
amazing! tyvm for providing more context!!! 👍

**Pull request merged**